rust-symcrypt/src/ecc/ecdh.rs

```diff
@@ -0,0 +1,253 @@
+//! EcDh functions. For further documentation please refer to symcrypt.h
+//!
+//! # Example
+//!
+//! ## EcDh with generated keys
+//!
+//! ```rust
+//! use symcrypt::ecc::{EcKey, CurveType, EcKeyUsage};
+//!
+//! // Set up 2 separate EcDH structs with public/private key pair attached.
+//!
+//! let key_1 = EcKey::generate_key_pair(CurveType::NistP256, EcKeyUsage::EcDh).unwrap();
+//! let key_2 = EcKey::generate_key_pair(CurveType::NistP256, EcKeyUsage::EcDh).unwrap();
+//!
+//! // Assert that the CurveType matches NistP256.
+//! assert_eq!(key_1.get_curve_type(), CurveType::NistP256);
+//! assert_eq!(key_2.get_curve_type(), CurveType::NistP256);
+//!
+//! // Get the public bytes from each EcDh struct generated.
+//! let public_bytes_1 = key_1.export_public_key().unwrap();
+//! let public_bytes_2 = key_2.export_public_key().unwrap();
+//!
+//! // Calculates secret agreements between private/public keys of two EcKey structs. The result
+//! // from each secret agreement should be the same.
+//! let ecdh_1_public =
+//! EcKey::set_public_key(CurveType::NistP256, &public_bytes_1.as_slice(), EcKeyUsage::EcDh).unwrap();
+//! let ecdh_2_public =
+//! EcKey::set_public_key(CurveType::NistP256, &public_bytes_2.as_slice(), EcKeyUsage::EcDh).unwrap();
+//!
+//! let secret_agreement_1 = key_1.ecdh_secret_agreement(ecdh_2_public).unwrap();
+//! let secret_agreement_2 = key_2.ecdh_secret_agreement(ecdh_1_public).unwrap();
+//!
+//! assert_eq!(secret_agreement_1, secret_agreement_2);
+//!
+//! ```
+//!
+use crate::ecc::{curve_to_num_format, EcKey};
+use crate::errors::SymCryptError;
+use std::vec;
+use symcrypt_sys;
+
+/// Impl for EcDh struct.
+impl EcKey {
+    /// `ecdh_secret_agreement()` returns a [`EcDhSecretAgreement`] that represents the secret agreement between the private key and the public key,
```

Comment:
>that represents the secret agreement between the private key and the public key [](http://example.com/codeflow?start=68&length=79)

This phrasing is kind of weird; I'd probably just say that it returns the agreed secret as a `Vec<u8>`

Reply:
Changed

Reply:
To clarify, this wasn't about the type, but about the language of the rest of the sentence. "The secret agreement between the private key and the public key" is not wrong, but it reads strangely to me, hence I'd just say that it "returns the agreed secret"